### PR TITLE
[Merged by Bors] - Lighthouse v0.2.0 (Medalla)

### PR DESCRIPTION
**Pull request by paulhauner**
## Issue Addressed

NA

## Proposed Changes

- Moves the git-based versioning we were doing into the `lighthouse_version` crate in `common`.
- Removes the `beacon_node/version` crate, replacing it with `lighthouse_version`.
- Bumps the version to `v0.2.0`.

## Additional Info

There are now two types of version string:

1. `const VERSION: &str = Lighthouse/v0.2.0-1419501f2+`
1. `version_with_platform() = Lighthouse/v0.2.0-1419501f2+/x86_64-linux`

(1) is handy cause it's a `const` and shorter. (2) has platform info so it's more useful. Note that the plus-sign (`+`) indicates the the git commit is dirty (it used to be `(modified)` but I had to shorten it to fit into graffiti).

These version strings are now included on:

- `lighthouse --version`
- `lcli --version`
- `curl localhost:5052/node/version`
- p2p messages when we communicate our version

You can update the version by changing this constant (version is not related to a `Cargo.toml`):

https://github.com/sigp/lighthouse/blob/b9ad7102d56f4827b8b74447c7121174a3e97bbc/common/lighthouse_version/src/lib.rs#L4-L15

### CODE REVIEW

File: beacon_node/src/cli.rs [hunk 1]
```diff
@@ -250,7 +244,7 @@ pub fn cli_app<'a, 'b>() -> App<'a, 'b> {
                 .long("graffiti")
                 .help("Specify your custom graffiti to be included in blocks.")
                 .value_name("GRAFFITI")
-                .default_value(DEFAULT_GRAFFITI)
+                .default_value(lighthouse_version::VERSION)
```
Comment (AgeManning) [inline beacon_node/src/cli.rs:247]: If we are going to do this, we should probably build a data type that ensures this is < `GRAFFITI_BYTES_LEN` so that it doesn't prevent clients from starting.

File: beacon_node/network/Cargo.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 [package]
 name = "network"
 version = "0.1.2"
-authors = ["Sigma Prime <contact@sigmaprime.io>"]
+authors = ["Age Manning <Age@AgeManning.com>"]
```
Comment (AgeManning) [inline beacon_node/network/Cargo.toml:4]: Can you revert these changes and others

File: beacon_node/src/cli.rs (hunk 1 above)
Comment (paulhauner) [reply beacon_node/src/cli.rs:247]: It only prevents it from starting if the user-supplied string is too long. If our in-built string is too long it truncates.

Comment (AgeManning) [reply beacon_node/src/cli.rs:247]: I think it errors if it's too long and the user doesn't specify

Comment (paulhauner) [reply beacon_node/src/cli.rs]: Yeah you were right, I fixed it.

**Comment by paulhauner** [comment]
All comments addressed

**Comment by michaelsproul** [comment]
If we could merge https://github.com/sigp/lighthouse/pull/1453 before this I would be grateful :)
